spec/controllers/companies_controller_spec.rb

```diff
@@ -2,17 +2,34 @@
 
 RSpec.describe CompaniesController, type: :controller do
   describe 'GET #index' do
-    it 'assigns all the companies to @companies' do
-      c1 = create(:company)
-      c2 = create(:company)
-      c3 = create(:company)
-      get :index
-      expect(assigns(:companies)).to eq([c1, c2, c3])
+    context 'no search params' do
+      it 'assigns all the companies to @companies' do
+        c1 = create(:company)
+        c2 = create(:company)
+        c3 = create(:company)
+        get :index
+        expect(assigns(:companies)).to eq([c1, c2, c3])
+      end
+
+      it "renders the index template" do
+        get :index
+        expect(response).to render_template("index.html.erb")
+      end
     end
 
-    it "renders the index template" do
-      get :index
-      expect(response).to render_template("index.html.erb")
+    context 'search params present' do
+      it 'assigns all the companies to @companies' do
+        c1 = create(:company, name: "fgsefe")
+        c2 = create(:company)
+        c3 = create(:company)
+        get :index, parameters: {query: ""}
+        expect(assigns(:companies)).to eq([c1, c2, c3])
```

Comment:
This isn't testing our search ability. I'm gonna merge it in and try to see why you were having problems and push the proper test up